src/ddprof_context_lib.c

```diff
@@ -20,19 +20,16 @@ DDRes ddprof_context_set(DDProfInput *input, DDProfContext *ctx) {
   memset(ctx, 0, sizeof(DDProfContext));
   setup_logger(input->log_mode, input->log_level);
 
-  for (int idx_watcher = 0; idx_watcher < input->num_watchers; ++idx_watcher) {
-    ctx->watchers[ctx->num_watchers] =
-        *(perfoptions_preset(input->watchers[idx_watcher]));
-    if (input->sampling_value[idx_watcher]) // override preset
-      ctx->watchers[ctx->num_watchers].sample_period =
-          input->sampling_value[idx_watcher];
-  }
-  ctx->num_watchers = input->num_watchers;
+  // Shallow copy of the watchers from the input object into the context.  This
+  // enables us to easily manage the lifetime of
+  ctx->num_watchers = 0;
+  for (int *i = &ctx->num_watchers; *i < input->num_watchers; ++*i)
```

Comment:
I find this loop hard to read ! Perhaps just using an integer counter and setting `cts->num_watchers = input->num_watchers` would be clearer.

Reply:
Not sure what I was thinking there?  Fixed.